RecipeApp/Modules/RecipeBook/View/RecipeBookViewController.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  RecipeBookViewController.swift
+//  RecipeApp
+//
+//  Created by Hadley Igoe on 2019-12-17.
+//  Copyright © 2019 Hadley Igoe. All rights reserved.
+//
+
+import UIKit
+
+final class RecipeBookViewController: UIViewController, ViewControllerCreatableWithViewModel {
+    
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var emptyView: UIView!
+    
+    var viewModel: RecipeBookViewModel!
+    weak var coordinator: RecipeBookCoordinator?
+       
+   static func createViewController() -> RecipeBookViewController {
+       let id = String(describing: self)
+       let storyboard = UIStoryboard(name: "RecipeBook", bundle: Bundle.main )
+       return storyboard.instantiateViewController(withIdentifier: id) as! Self
+   }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        updateUI()
+        viewModel.updateHandler = { [weak self] in self?.updateUI() }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        emptyView.isHidden = false
+        tableView.isHidden = true
+        viewModel.viewWillAppear()
+        setupTableView()
+    }
+    
+    private func setupTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        let headerNib = UINib.init(nibName: "RecipeBookHeaderView", bundle: Bundle.main)
+        tableView.register(headerNib, forHeaderFooterViewReuseIdentifier: "RecipeBookHeaderView")
+        tableView.rowHeight = 100
+        tableView.separatorStyle = .none
+        tableView.backgroundColor = .white
+        updateUI()
+    }
+    
+    private func updateUI() {
+        tabBarController?.tabBar.isHidden = false
```

Comment:
This line never changes....updateUI should only update things in the view that change.